frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseOverrideForkPointAction.java

```diff
@@ -0,0 +1,144 @@
+package com.virtuslab.gitmachete.frontend.actions.base;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+import static org.checkerframework.checker.i18nformatter.qual.I18nConversionCategory.GENERAL;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsException;
+import com.intellij.openapi.vfs.VirtualFile;
+import git4idea.config.GitConfigUtil;
+import io.vavr.collection.List;
+import lombok.CustomLog;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.i18nformatter.qual.I18nFormat;
+
+import com.virtuslab.gitmachete.backend.api.IGitMacheteBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteCommit;
+import com.virtuslab.gitmachete.backend.api.SyncToParentStatus;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.OverrideForkPointDialog;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyGitMacheteRepository;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyProject;
+import com.virtuslab.logger.IEnhancedLambdaLogger;
+
+@CustomLog
+public abstract class BaseOverrideForkPointAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IExpectsKeyProject,
+      IExpectsKeyGitMacheteRepository,
+      ISyncToParentStatusDependentAction {
+
+  @Override
+  public IEnhancedLambdaLogger log() {
+    return LOG;
+  }
+
+  @Override
+  public @I18nFormat({}) String getActionName() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.action-name");
+  }
+
+  @Override
+  public @I18nFormat({}) String getDescriptionActionName() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.description-action-name");
+  }
+
+  @Override
+  public @I18nFormat({GENERAL, GENERAL}) String getEnabledDescriptionFormat() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.description");
+  }
+
+  @Override
+  public String getCurrentBranchText() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.text.current-branch");
+  }
+
+  @Override
+  public List<SyncToParentStatus> getEligibleStatuses() {
+    return List.of(SyncToParentStatus.InSyncButForkPointOff);
```

Comment:
I know my use cases might be weird... but after using a snapshot build from this PR for a couple of hours, I actually needed to override fork point for a red-edge branch... so TBH including red branches won't hurt.

I don't remember a single case, though, when I needed to override it for a green edge, let alone gray edge... so probably these can be skipped.

Reply:
Awww crap... overriding for red edge doesn't make much sense until #449 is implemented... ignore the above comment then